### PR TITLE
fix: skip solana unsupported transaction version

### PR DESCRIPTION
**Pull request by ws4charlie**
# Description

There are two types of Solana transactions in Solana network: `legacy` and `0`. Current zetaclient implementation can only pull `legacy` inbound transaction and returns error on version `0`. This will blocks inbound observation if user send a deposit tx to gateway using version `0`.

The fix:

- [x] Allow version `0` in the RPC query `getTransaction`.
- [x] Skip any unsupported transaction type, so inbound won't be impact if Solana protocol adds new transaction type at some point in the future.

# How Has This Been Tested?

- [x] Tested CCTX in localnet
- [ ] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Release Notes

- **New Features**
  - Introduced a new "Unreleased" section in the changelog for better organization of upcoming changes.
  - Added functionality to handle unsupported Solana transaction versions, preventing disruptions in transaction processing.

- **Bug Fixes**
  - Enhanced error handling for unsupported transaction versions, ensuring smoother transaction retrieval.

- **Tests**
  - Added a new live test to verify transaction retrieval based on version support, improving test coverage for RPC functionality.

### CODE REVIEW

File: zetaclient/chains/solana/rpc/rpc.go [hunk 1]
```diff
@@ -18,6 +19,10 @@ const (
 	// RPCAlertLatency is the default threshold for RPC latency to be considered unhealthy and trigger an alert.
 	// The 'HEALTH_CHECK_SLOT_DISTANCE' is default to 150 slots, which is 150 * 0.4s = 60s
 	RPCAlertLatency = time.Duration(60) * time.Second
+
+	// JSONRPCErrorCodeUnsupportedTransactionVersion
```
Comment (lumtis) [inline zetaclient/chains/solana/rpc/rpc.go:23]:
```suggestion
	// ErrorCodeUnsupportedTransactionVersion
```

File: zetaclient/chains/solana/rpc/rpc.go [hunk 2]
```diff
@@ -122,6 +127,25 @@ func GetSignaturesForAddressUntil(
...
+	signature solana.Signature,
+	maxTxVersion *uint64,
+) (*rpc.GetTransactionResult, error) {
+	txResult, err := client.GetTransaction(ctx, signature, &rpc.GetTransactionOpts{
+		MaxSupportedTransactionVersion: maxTxVersion,
+	})
+
+	// skip unsupported transaction version error
+	if err != nil && strings.Contains(err.Error(), ErrorCodeUnsupportedTransactionVersion) {
+		return nil, nil
```
Comment (lumtis) [inline zetaclient/chains/solana/rpc/rpc.go:143]: Can we rather return a boolean unsupported, or a error that can be filtered? I think would be cleaner than returning a nil

Comment (ws4charlie) [reply zetaclient/chains/solana/rpc/rpc.go:143]: Returning an additional boolean `skip` looks more readable.

Comment (ws4charlie) [reply zetaclient/chains/solana/rpc/rpc.go]: added boolean https://github.com/zeta-chain/node/pull/3206/commits/4f0e1716e43c045e4a6427f0df2738e231008429

File: zetaclient/chains/solana/rpc/rpc.go (hunk 1 above)
Comment (ws4charlie) [reply zetaclient/chains/solana/rpc/rpc.go]: https://github.com/zeta-chain/node/pull/3206/commits/4f0e1716e43c045e4a6427f0df2738e231008429

**Review by lumtis — APPROVED** [review]

File: zetaclient/chains/solana/rpc/rpc.go [hunk 3]
```diff
@@ -122,6 +127,25 @@ func GetSignaturesForAddressUntil(
 	return allSignatures, nil
 }
 
+// GetTransactionWithMaxVersion fetches a transaction with the given signature and max version.
+func GetTransactionWithMaxVersion(
```
Comment (swift1337) [inline zetaclient/chains/solana/rpc/rpc.go:131]: Let's hide the complexity of specifying max tx version as this rpc package is used only in zetaclient
```go
var ErrUnsupportedTxVersion = errors.New("unsupported tx version")

// GetTransaction fetches tx by the signature. Might also return ErrUnsupportedTxVersion
func GetTransaction(ctx context.Context, c interfaces.SolanaRPCClient, sig solana.Signature) (*rpc.GetTransactionResult, error) {

 // ...
}

```

File: zetaclient/chains/solana/rpc/rpc.go [hunk 4]
```diff
@@ -18,6 +19,10 @@ const (
 	// RPCAlertLatency is the default threshold for RPC latency to be considered unhealthy and trigger an alert.
 	// The 'HEALTH_CHECK_SLOT_DISTANCE' is default to 150 slots, which is 150 * 0.4s = 60s
 	RPCAlertLatency = time.Duration(60) * time.Second
+
+	// ErrorCodeUnsupportedTransactionVersion
+	// see: https://github.com/solana-labs/solana/blob/master/rpc/src/rpc.rs#L7276
+	ErrorCodeUnsupportedTransactionVersion = "-32015"
```
Comment (swift1337) [inline zetaclient/chains/solana/rpc/rpc.go:25]: let's make this unexported.

File: zetaclient/chains/solana/rpc/rpc_live_test.go [hunk 5]
```diff
@@ -17,11 +17,44 @@ func Test_SolanaRPCLive(t *testing.T) {
...
+		)
+		require.NoError(t, err)
+		require.NotNil(t, txResult)
+		require.False(t, skip)
+	})
+
+	t.Run("should skip the transaction if the version is not supported", func(t *testing.T) {
+		ctx := context.Background()
+		txResult, skip, err := rpc.GetTransactionWithMaxVersion(ctx, client, txSig, nil)
+		require.NoError(t, err)
```
Comment (swift1337) [inline zetaclient/chains/solana/rpc/rpc_live_test.go:52]: like so

`require.ErrorsIs(err, ErrUnsupportedTxVersion)`

File: zetaclient/chains/solana/observer/inbound.go [hunk 6]
```diff
@@ -100,17 +100,28 @@ func (ob *Observer) ObserveInbound(ctx context.Context) error {
...
 			}
 
-			// filter inbound events and vote
-			err = ob.FilterInboundEventsAndVote(ctx, txResult)
-			if err != nil {
-				// we have to re-scan this signature on next ticker
-				return errors.Wrapf(err, "error FilterInboundEventAndVote for chain %d sig %s", chainID, sigString)
+			switch {
+			case skip:
+				ob.Logger().Inbound.Warn().Msgf("ObserveInbound: skip unsupported transaction sig %s", sigString)
```
Comment (swift1337) [inline zetaclient/chains/solana/observer/inbound.go:117]: Let's replace all `%s` with structured fields e.g. `"tx.signature"`

File: zetaclient/chains/solana/rpc/rpc.go (hunk 3 above)
Comment (ws4charlie) [reply zetaclient/chains/solana/rpc/rpc.go:131]: The version is used by unit test to verify different behaviors [here](https://github.com/zeta-chain/node/blob/fix-solana-unsupported-tx-version-0/zetaclient/chains/solana/rpc/rpc_live_test.go#L51)

File: zetaclient/chains/solana/rpc/rpc_live_test.go (hunk 5 above)
Comment (ws4charlie) [reply zetaclient/chains/solana/rpc/rpc_live_test.go:52]: Not sure what you meant. It is no error in the test.

**Pull request merged**